### ISSUE TITLE
Better panic message for filling with NaN

### ISSUE THREAD

**Issue by IQuick143** [davehadley/ndhistogram#33]
When the user attempts to fill the histogram (with an uniform axis) with a NaN value, they obtain the panic message:
```
thread 'main' panicked at 'bin number can always be converted to a valid usize', /[registry path]/ndhistogram-0.9.0/src/axis/uniform.rs:115:18
```
Which happened because the expectation that the bin number can be converted to an usize is violated for a NaN input value. (Is there perhaps another invalid input value?)

Code to reproduce:
```
use ndhistogram::{ndhistogram, axis::Uniform, Histogram};

fn main() {
	let mut hist = ndhistogram!(Uniform::new(100, 0.0, 1.0));
	hist.fill(&f32::NAN);
}
```

The message was fairly confusing to me to debug when I encountered it in the wild, initially I thought there was an overflow during the conversion (so I started searching for gigantic inputs) and I did not realize that the dataset contained a NaN.

Panicking on a NaN value seems appropriate, rather than returning a Result, but I think it would be helpful for future users if there was a more descriptive message (at least when run in a debug build) telling them an invalid value was passed, so they know right away what went wrong and where to start searching.

Sorry if this is a nit-picky issue, I just thought it would've saved me a bunch of time and the same would apply for future users running into this.

**Comment by davehadley** [comment]
Thank you for the report. Not nit-picky at all! I think this is a flaw in the library. At least for axes with overflow bins, NaN should probably go into the overflow. Certainly panic seems like the wrong thing to do.

**Comment by davehadley** [comment]
This is fixed in #35 and will be included in the next release. Filling with NAN should no longer panic. These values will go into an overflow bin if it exists. If the axis does not have overflow bins, NaN values will map to any bin and so no bin will be filled.
